allennlp/common/file_utils.py

```diff
@@ -577,6 +580,17 @@ class TensorCache(MutableMapping[str, Tensor], ABC):
     runs at the same time.
     """
 
+    def __new__(cls, filename: Union[str, PathLike], *, read_only: bool = False, **kwargs):
+        # This mechanism makes sure we re-use open lmdb file handles. Lmdb has a problem when the same file is
+        # opened by the same process multiple times. This is our workaround.
+        filename = str(filename)
```

Comment:
We should probably normalize `filename` to an absolute path here, right?

Reply:
I'll do you one better and do it by inode: 153bade2b61abb54c17807934e59ee11cfaf4957

That way even symlinks and hard links work correctly.

Reply:
💯